parquet-column/src/main/java/org/apache/parquet/schema/LogicalTypeAnnotation.java

```diff
@@ -36,42 +36,152 @@
 import org.apache.parquet.format.TimeType;
 import org.apache.parquet.format.TimestampType;
 
+import java.util.List;
 import java.util.Objects;
 
-public interface LogicalTypeAnnotation {
+public abstract class LogicalTypeAnnotation {
+  public enum LogicalTypes {
```

Comment:
This enum is used only for parsing/printing and we don't want the users to really use them. So, I would suggest using a name that suggests its use e.g. `LogicalTypeParseHelper`?
Also, it would be nice if we could annotate/comment that this one is not part of the public API.